packages/tenant-readmodel-writer/src/tenantConsumerService.ts

```diff
@@ -50,11 +53,67 @@ export async function handleMessage(
           }
         )
     )
-    .with({ type: "SelfcareMappingCreated" }, async (_msg) => {
-      logger.info("TODO");
+    .with({ type: "SelfcareMappingCreated" }, async (msg) => {
+      await tenants.updateOne(
+        {
+          "data.id": msg.stream_id,
+          "metadata.version": { $lt: msg.version },
+        },
+        {
+          $set: {
+            "data.selfcareId": msg.data.selfcareId,
+            "metadata.version": msg.version,
+          },
+        }
+      );
+    })
+    .with({ type: "SelfcareMappingDeleted" }, async (msg) => {
+      await tenants.updateOne(
+        {
+          "data.id": msg.stream_id,
+          "metadata.version": { $lt: msg.version },
+        },
+        {
+          $set: {
+            "data.selfcareId": undefined,
+            "metadata.version": msg.version,
+          },
+        }
+      );
     })
-    .with({ type: "SelfcareMappingDeleted" }, async (_msg) => {
-      logger.info("TODO");
+    .with({ type: "TenantMailAdded" }, async (msg) => {
+      await tenants.updateOne(
+        {
+          "data.id": msg.stream_id,
+        },
+        {
+          $set: {
+            data: msg.data.tenant ? fromTenantV1(msg.data.tenant) : undefined,
+            metadata: {
+              version: msg.version,
+            },
+          },
+        }
```

Comment:
```suggestion
        },
        { upsert: true }
```
Same

Reply:
I don't think so. If we are adding an email for a tenant, we already have that tenant in readmodel.